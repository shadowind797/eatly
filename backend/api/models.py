from django.db import models
from django.contrib.auth.models import AbstractUser


class User(AbstractUser):
    username = models.CharField(max_length=120, unique=True)
    password = models.CharField(max_length=120)
    phone = models.CharField(max_length=120, blank=True)
    status = models.ForeignKey('Status', on_delete=models.PROTECT, default=5)
    is_banned = models.BooleanField(default=False)
    ban_reason = models.TextField(blank=True)
    passwd_change_link = models.CharField(max_length=120, blank=True)

    def __str__(self):
        return self.username


class Item(models.Model):
    title = models.CharField(max_length=120, unique=True)
    description = models.TextField()
    price = models.FloatField()
    photo = models.ImageField(upload_to='img')
    category = models.ForeignKey('Category', on_delete=models.SET_NULL, null=True, blank=True)
    restaurant = models.ForeignKey('Restaurant', on_delete=models.SET_NULL, null=True, blank=True)
    rating = models.FloatField()

    def __str__(self):
        return self.title


class Restaurant(models.Model):
    name = models.CharField(max_length=120, unique=True)
    address = models.CharField(max_length=120)
    rating = models.FloatField()
    image = models.ImageField(upload_to='img/rests', null=True)
    category = models.ForeignKey('RestaurantCat', on_delete=models.SET_NULL, null=True, blank=True)

    def __str__(self):
        return self.name


class Address(models.Model):
    house_address = models.CharField(max_length=120)
    entrance = models.CharField(max_length=120)
    floor = models.CharField(max_length=120)
    flat = models.CharField(max_length=120)
    owner = models.ForeignKey('User', on_delete=models.CASCADE)
    active = models.BooleanField(default=False)

    def __str__(self):
        return self.house_address


class Order(models.Model):
    user = models.ForeignKey('User', on_delete=models.SET_NULL, null=True)
    payment = models.ForeignKey('Payments', on_delete=models.SET_NULL, null=True)
    address = models.ForeignKey('Address', on_delete=models.SET_NULL, null=True)
    total = models.FloatField()
    status = models.ForeignKey('OrderStatus', on_delete=models.SET_NULL, null=True)
    restaurant = models.ForeignKey('Restaurant', on_delete=models.SET_NULL, null=True)
    coupon = models.ForeignKey('Coupon', on_delete=models.SET_NULL, null=True)
    comment = models.TextField(null=True, blank=True)

    def __str__(self):
        return self.user


class OrderStatus(models.Model):
    name = models.CharField(max_length=120, unique=True)


class Coupon(models.Model):
    title = models.CharField(max_length=120, unique=True)
    valid_from = models.DateField(auto_now_add=True)
    valid_to = models.DateField()
    is_valid = models.BooleanField(default=True)
    value = models.FloatField()
    times_activated = models.IntegerField(default=0)
    enabled_times = models.IntegerField()
    category = models.ForeignKey('CouponCategory', on_delete=models.CASCADE)

    def __str__(self):
        return self.title


class CouponCategory(models.Model):
    name = models.CharField(max_length=120, unique=True)

    def __str__(self):
        return self.name


class Payments(models.Model):
    number = models.CharField(max_length=120, unique=True)
    date_to = models.CharField(max_length=100)
    cvv = models.IntegerField()
    owner = models.ForeignKey('User', on_delete=models.CASCADE)
    active = models.BooleanField(default=False)
    name = models.CharField(max_length=120, default="")

    def __str__(self):
        return self.number


class CartItem(models.Model):
    item = models.ForeignKey('Item', on_delete=models.CASCADE)
    quantity = models.IntegerField(default=1)
    owner = models.ForeignKey('User', on_delete=models.CASCADE)

    def __str__(self):
        return self.item.title


class Reviews(models.Model):
    author = models.ForeignKey('User', on_delete=models.CASCADE)
    time = models.CharField(max_length=120)
    text = models.TextField()
    rating = models.FloatField()
    created_at = models.DateTimeField(auto_now_add=True)

    def __str__(self):
        return self.author.username


class RestaurantCat(models.Model):
    name = models.CharField(max_length=120, unique=True)

    def __str__(self):
        return self.name


class Category(models.Model):
    name = models.CharField(max_length=120, unique=True)

    def __str__(self):
        return self.name


class Status(models.Model):
    name = models.CharField(max_length=120)

    def __str__(self):
        return self.name
