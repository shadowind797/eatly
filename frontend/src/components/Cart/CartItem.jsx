import {useEffect, useState} from "react";
import api from "../../api.js";

function CartItem({cartItem}){
    const [dish, setDish] = useState({});
    const [quantity, setQuantity] = useState(cartItem.quantity);
    const [deleted, setDeleted] = useState(false);

    useEffect(() => {
        getItems()
    }, []);

    const getItems = () => {
        api
            .get("api/items/", {params: {id: cartItem.item}})
            .then((res) => res.data)
            .then((data) => {
                data.map((item) => {
                    if (item.id === cartItem.item) {
                        setDish(item)
                    }
                })
            })
            .catch((err) => alert(err));
    }

    const deleteItem = () => {
        api.delete("api/items/cart/delete",
            {params: {id: cartItem.id}})
            .then((res) => {
                if (res.status === 202) {}
                else if (res.status === 404) {}
            }).catch((err) => {});
        setDeleted(true)
    }

    const addQuantity = () => {
        api.post("api/items/cart/add", {item: dish.id, quantity: quantity + 1},
                                                    {params: {method: "addQuant"}})
                                                                .then((res) => {
            if (res.status === 201) {} else {
                alert("Failed to create item");
            }
        }).catch((err) => {});
    }

    const removeQuantity = () => {
        if (quantity > 1) {
            api.post("api/items/cart/add", {item: dish.id, quantity: quantity - 1},
                {params: {method: "addQuant"}})
                .then((res) => {
                    if (res.status === 201) {
                    } else {
                        alert("Failed to create item");
                    }
                }).catch((err) => {
            });
        } else {
            deleteItem()
        }
    }

    if (deleted === false) {
        return (
            <div>
                <img src={dish.photo} alt=""/>
                <p>{quantity}</p>
                <button onClick={() => {
                    setQuantity(quantity + 1)
                    addQuantity()
                }}>addItem
                </button>
                <button onClick={() => {
                    setQuantity(quantity - 1)
                    removeQuantity()
                }}>removeItem
                </button>
            </div>
        )
    } else {
        return (<p></p>)
    }
}

export default CartItem