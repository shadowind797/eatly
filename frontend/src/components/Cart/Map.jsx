import GoogleMapReact from "google-map-react"
import {useState, useEffect} from "react";

const Marker = props => {
    return <div className="SuperAwesomePin">Marker</div>
}

function Map({address}) {
    const [marker, setMarker] = useState(null);

    const onGoogleApiLoaded = ({map, maps}) => {
        const geocoder = new maps.Geocoder();
        geocoder.geocode({address}, (results, status) => {
            if (status === "OK" && results[0]) {
                const marker = new maps.Marker({
                    position: results[0].geometry.location,
                });
                marker.setMap(map);
                setMarker(marker);
            } else {
                console.error("Geocode was not successful for the following reason: " + status);
            }
        });
    };

    return (
        <div id="map">
            <GoogleMapReact
                bootstrapURLKeys={{key: import.meta.env.VITE_MAP_KEY, libraries: ["places"]}}
                defaultCenter={{lat: 41.316234975275385, lng: 69.24842619026026}}
                defaultZoom={address ? 14 : 11}
                options={{
                    zoomControl: false,
                    streetViewControl: false,
                    mapTypeControl: false,
                    scaleControl: false,
                    rotateControl: false,
                    fullscreenControl: false,
                    keyboardShortcuts: false,
                    clickableIcons: false,
                    scrollwheel: false,
                    gestureHandling: 'none',
                }}
            >
                {address && <Marker lat={onGoogleApiLoaded.lat} lng={onGoogleApiLoaded.lng}/>}
            </GoogleMapReact>
        </div>
    )
}

export default Map;