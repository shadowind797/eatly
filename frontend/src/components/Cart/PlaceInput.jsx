import PlacesAutocomplete, {geocodeByAddress, getLatLng} from "react-places-autocomplete";
import React, {useState} from "react";

const PlaceInput = ({finalAddress}) => {
    const [address, setAddress] = useState('');

    const handleChange = address => {
        setAddress(address);
    };

    const handleSelect = address => {
        setAddress(address)
        finalAddress(address)
    };

    return (
        <PlacesAutocomplete
            value={address}
            onChange={handleChange}
            onSelect={handleSelect}
        >
            {({getInputProps, suggestions, getSuggestionItemProps, loading}) => (
                <div style={{width: "470px", position: "relative", zIndex: "16"}}>
                    <input
                        {...getInputProps({
                            placeholder: 'Building',
                            className: 'location-search-input',
                            style: {width: "96%"},
                        })}
                    />
                    <div className="autocomplete-dropdown-container"
                         style={{
                             position: "absolute",
                             zIndex: "16",
                             flexDirection: "column",
                             marginTop: "57px",
                             backgroundColor: "#ffffff",
                             border: "1px solid #c5c5c5",
                         }}>
                        {loading && <div>Loading...</div>}
                        {suggestions.map(suggestion => {
                            const className = suggestion.active
                                ? 'suggestion-item--active'
                                : 'suggestion-item';
                            return (
                                <div
                                    {...getSuggestionItemProps(suggestion, {
                                        className,
                                        style: {padding: "12px", cursor: "pointer"},
                                    })}
                                >
                                    <span>{suggestion.description}</span>
                                </div>
                            );
                        })}
                    </div>
                </div>
            )}
        </PlacesAutocomplete>
    );
};

export default PlaceInput;