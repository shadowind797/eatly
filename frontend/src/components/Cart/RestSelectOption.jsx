/* eslint-disable react/prop-types */
import { components } from "react-select";

const RestSelectOption = (props) => {
  return (
    <components.Option {...props}>
      <img
        src={`${import.meta.env.VITE_API_URL}/media${props.data.logo}`}
        alt=""
        style={{ width: 20, height: 20, marginRight: 10 }}
      />
      {props.data.label}
    </components.Option>
  );
};

export default RestSelectOption;
