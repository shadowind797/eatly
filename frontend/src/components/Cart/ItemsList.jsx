import CartItem from "./CartItem.jsx";
import PropTypes from "prop-types"

ItemsList.propTypes = {
  items: PropTypes.array.isRequired,
  cartItems: PropTypes.array.isRequired,
  onChange: PropTypes.func.isRequired,
}

function ItemsList({items, cartItems, onChange}) {

    return (
        <div id='cart-items-list' className='container'>
            {items.map((item) => <CartItem dish={item} cartItem={cartItems.find(ci => ci.item_id === item.id)}
                                           onChange={onChange} key={item.id}/>)}
        </div>
    )
}

export default ItemsList;
