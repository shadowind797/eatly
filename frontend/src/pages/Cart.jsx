import Header from "../components/Header.jsx";
import ItemsList from "../components/Cart/ItemsList.jsx";
import BaseHeader from "../components/BaseHeader.jsx";
import MakeOrder from "../components/Cart/MakeOrder.jsx";
import {useEffect, useState} from "react";
import api from "../api.js";

function Cart() {
    const [cartItems, setCartItems] = useState([]);
    const [total, setTotal] = useState(0);


    useEffect(() => {
        getCartItems()
    }, []);

    const getCartItems = () => {
        api
            .get("api/items/cart/")
            .then((res) => res.data)
            .then((data) => {
                setCartItems(data)
                getTotal(data)
            })
    }

    const getTotal = (cartI) => {
        api
            .post("api/items/", {data: {items: cartI, method: "for_total"}})
            .then((res) => res.data)
            .then((data) => setTotal(data))
            .catch((err) => alert(err));
    }


    if (cartItems.length > 0 && total.total > 0) {
        return (
            <div id='cart'>
                <BaseHeader />
                <Header />
                <div id="main">
                    <ItemsList items={cartItems} onChange={() => {getCartItems()}}/>
                    <MakeOrder itemsC={cartItems} subtotal={total.total}/>
                </div>
            </div>
        )
    }
}

export default Cart;
