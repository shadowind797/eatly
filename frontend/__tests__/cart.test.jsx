import { render, fireEvent, waitFor, act } from "@testing-library/react";
import api from "../src/api.js";
import { MemoryRouter as Router } from "react-router-dom";
import Cart from "../src/pages/Cart.jsx";

jest.mock("../src/api");

describe("Cart component", () => {
  const cartItem = {
    rests: [
      {
        id: 3,
        name: "KFC",
      },
      {
        id: 5,
        name: "Maurice",
      },
    ],
    items: [
      {
        id: 5,
        restaurant_id: 3,
        title: "Swe Dish",
        price: 19.0,
        photo: "img/food/SweDish.png",
      },
      {
        id: 17,
        restaurant_id: 5,
        title: "Fried Dorado",
        price: 15.0,
        photo: "img/food/foodimg1.png",
      },
    ],
    cart_items: [
      {
        id: 116,
        quantity: 3,
        item_id: 5,
      },
      {
        id: 125,
        quantity: 3,
        item_id: 17,
      },
    ],
  };

  const user = [
    {
      id: 1,
      username: "shadowind",
      email: "shadowind797@gmail.com",
      phone: "+375666999",
      status: 1,
      is_banned: false,
      ban_reason: "",
      first_name: "",
    },
  ];
  const addresses = [
    {
      id: 2,
      house_address: "Mirobod street 12, Tashkent, UZ",
      entrance: "2",
      floor: "4",
      flat: "34",
    },
  ];

  beforeEach(() => {
    api.get.mockImplementation((url) => {
      if (url.includes("api/items/cart/data/")) {
        return Promise.resolve({ status: 200, data: cartItem });
      }
      if (url.includes("api/user/")) {
        return Promise.resolve({ status: 200, data: user });
      }
      if (url.includes("api/address/")) {
        return Promise.resolve({ status: 200, data: addresses });
      }
      return Promise.reject(new Error("Not found"));
    });
    api.post.mockImplementation((url) => {
      if (url.includes("api/items/cart/add")) {
        return Promise.resolve({ status: 201 });
      } else if (url.includes("api/items/")) {
        return Promise.resolve({ status: 200, data: { total: 12.99 } });
      }
      if (url.includes("api/coupon/")) {
        return Promise.resolve({ status: 200, data: { value: 0.9 } });
      }
      if (url.includes("api/address/add/")) {
        return Promise.resolve({ status: 201 });
      }
      return Promise.reject(new Error("Not found"));
    });
    api.delete.mockImplementation((url) => {
      if (url.includes("api/items/cart/delete")) {
        return Promise.resolve({ status: 202 });
      }
      return Promise.reject(new Error("Not found"));
    });
  });

  it("renders without crashing", async () => {
    await act(async () => {
      render(
        <Router>
          <Cart />
        </Router>
      );
    });
  });

  it("matches the snapshot", async () => {
    let asFragment, getByTestId, getByText;
    await act(async () => {
      const rendered = render(
        <Router>
          <Cart />
        </Router>
      );
      asFragment = rendered.asFragment;
      getByTestId = rendered.getByTestId;
      getByText = rendered.getAllByText;
    });

    await waitFor(() => getByTestId("minus"));

    expect(asFragment()).toMatchSnapshot();

    await waitFor(() => {
      expect(getByText("$65.97")[0]).toBeTruthy();
    });
  });

  it("updates total when quantity changed", async () => {
    let getByText, getByTestId, asFragment;
    await act(async () => {
      const rendered = render(
        <Router>
          <Cart />
        </Router>
      );
      getByText = rendered.getAllByText;
      getByTestId = rendered.getByTestId;
      asFragment = rendered.asFragment;
    });

    await waitFor(() => getByTestId("minus"));

    fireEvent.click(getByTestId("minus"));
    await waitFor(() => {
      expect(api.get).toHaveBeenCalledWith("api/items/cart/data/");
    });

    fireEvent.click(getByTestId("plus"));
    await waitFor(() => {
      expect(api.get).toHaveBeenCalledWith("api/items/cart/data/");
    });
  });

  it("updates total when coupon applied", async () => {
    let getByText, getByTestId, asFragment, getByPlaceholderText;
    await act(async () => {
      const rendered = render(
        <Router>
          <Cart />
        </Router>
      );
      getByText = rendered.getAllByText;
      getByTestId = rendered.getByTestId;
      getByPlaceholderText = rendered.getByPlaceholderText;
      asFragment = rendered.asFragment;
    });

    await waitFor(() => getByTestId("minus"));

    fireEvent.change(getByPlaceholderText("Apply Coupon"), {
      target: { value: "TESTCOUPON" },
    });

    fireEvent.click(getByText("Apply")[0]);
    await waitFor(() => {
      expect(getByText("-$6.60")[0]).toBeTruthy();
    });
  });

  it("adds new user address", async () => {
    let getByText, getByTestId, asFragment, getByPlaceholderText, queryByTestId;
    await act(async () => {
      const rendered = render(
        <Router>
          <Cart test={true} />
        </Router>
      );
      getByText = rendered.getAllByText;
      getByTestId = rendered.getByTestId;
      getByPlaceholderText = rendered.getByPlaceholderText;
      queryByTestId = rendered.queryByTestId;
      asFragment = rendered.asFragment;
    });

    await waitFor(() => getByTestId("minus"));

    fireEvent.click(getByText("Add address")[0]);
    fireEvent.change(getByPlaceholderText("Building"), {
      target: { value: "Mirobod street 12, Tashkent, UZ" },
    });
    fireEvent.change(getByPlaceholderText("Entrance"), {
      target: { value: "2" },
    });
    fireEvent.change(getByPlaceholderText("Floor"), { target: { value: "4" } });
    fireEvent.change(getByPlaceholderText("Flat"), { target: { value: "34" } });

    fireEvent.click(getByText("Add address")[0]);

    await waitFor(() => {
      expect(getByText("$59.97")[0]).toBeTruthy();
    });
  });

  it("handles input errors", async () => {
    let getByText, getByTestId, asFragment, getByPlaceholderText, queryByTestId;
    await act(async () => {
      const rendered = render(
        <Router>
          <Cart />
        </Router>
      );
      getByText = rendered.getAllByText;
      getByTestId = rendered.getByTestId;
      getByPlaceholderText = rendered.getByPlaceholderText;
      queryByTestId = rendered.queryByTestId;
      asFragment = rendered.asFragment;
    });

    await waitFor(() => getByTestId("minus"));

    fireEvent.click(getByText("Complete order")[0]);

    await waitFor(() =>
      expect(getByText("Please enter your name")).toBeTruthy()
    );
    fireEvent.change(getByPlaceholderText("How courier'll call you?"), {
      target: { value: "Peter" },
    });

    fireEvent.click(getByText("Complete order")[0]);

    await waitFor(() => expect(getByText("Address required")).toBeTruthy());

    const selectComponent = queryByTestId("address-select");
    fireEvent.keyDown(selectComponent.childNodes[1], { key: "ArrowDown" });
    await waitFor(
      () =>
        getByText(
          "Mirobod street 12, Tashkent, UZ, 2 ent., 4 floor, flat 34"
        )[0]
    );
    fireEvent.click(
      getByText("Mirobod street 12, Tashkent, UZ, 2 ent., 4 floor, flat 34")[0]
    );
  });

  it("selects restaurant to deliver from", async () => {
    const { queryByTestId, getByTestId, getByText, asFragment } = render(
      <Router>
        <Cart />
      </Router>
    );

    await waitFor(() => getByTestId("minus"));

    const selectComponent = queryByTestId("rest-select");
    expect(asFragment()).toMatchSnapshot();

    fireEvent.keyDown(selectComponent.childNodes[0], { key: "ArrowDown" });
    await waitFor(() => getByText("Maurice"));
    fireEvent.click(getByText("Maurice"));

    await waitFor(() => {
      expect(getByText("$15.99")).toBeTruthy();
    });
  });
});
