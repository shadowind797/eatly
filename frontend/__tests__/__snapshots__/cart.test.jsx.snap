// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart component matches the snapshot 1`] = `
<DocumentFragment>
  <div
    id="cart"
  >
    <div
      class="base-header"
    >
      <img
        alt=""
        id="load-img"
        src="test-file-stub"
      />
    </div>
    <div
      id="main"
      style=""
    >
      <div
        class="container"
        id="cart-items-list"
      >
        <div
          class="item"
        >
          <button
            class="cross"
            data-testid="cross"
          >
            <img
              alt=""
              src="test-file-stub"
            />
          </button>
          <div
            class="main"
          >
            <img
              alt=""
              src="undefined/media/img/food/SweDish.png"
            />
            <div
              class="info"
            >
              <h4>
                Swe Dish
              </h4>
              <h5>
                $19.99
              </h5>
            </div>
          </div>
          <div
            class="quant"
          >
            <button
              data-testid="minus"
            >
              <img
                alt=""
                src="test-file-stub"
              />
            </button>
            <div>
              <p>
                3
              </p>
            </div>
            <button
              data-testid="plus"
            >
              <img
                alt=""
                src="test-file-stub"
              />
            </button>
          </div>
        </div>
      </div>
      <div
        style="display: block;"
      >
        <div
          id="order"
        >
          <div
            id="costs"
          >
            <p
              class="error"
              style="display: none;"
            >
              Invalid coupon
            </p>
            <form>
              <div
                id="input"
              >
                <img
                  alt=""
                  src="test-file-stub"
                />
                <input
                  placeholder="Apply Coupon"
                  type="text"
                  value=""
                />
              </div>
              <button
                type="submit"
              >
                Apply
              </button>
            </form>
            <h6>
              Subtotal: 
              <span>
                $59.97
              </span>
            </h6>
            <h6>
              Delivery: 
              <span>
                $6.00
              </span>
            </h6>
            <h6
              style="display: none;"
            >
              Coupon:
              <span
                style="color: green;"
              >
                -$0.00
              </span>
            </h6>
            <h6
              class="total"
            >
              Total: 
              <span>
                $65.97
              </span>
            </h6>
          </div>
          <div
            id="pay"
          >
            <div
              id="userInfo"
            >
              <div
                id="address"
              >
                <div
                  id="nameDiv"
                >
                  <input
                    autocomplete="off"
                    id="name"
                    placeholder="How courier'll call you?"
                    type="text"
                    value=""
                  />
                </div>
                <div
                  id="phoneDiv"
                >
                  <input
                    autocomplete="off"
                    id="name"
                    placeholder="Phone number"
                    type="phone"
                    value="+375666999"
                  />
                </div>
              </div>
              <div
                data-testid="rest-select"
                id="restDiv"
              >
                <div
                  class=" css-b62m3t-container"
                >
                  <span
                    class="css-1f43avz-a11yText-A11yText"
                    id="react-select-4-live-region"
                  />
                  <span
                    aria-atomic="false"
                    aria-live="polite"
                    aria-relevant="additions text"
                    class="css-1f43avz-a11yText-A11yText"
                    role="log"
                  />
                  <div
                    class=" css-185hrlv-control"
                  >
                    <div
                      class=" css-hlgwow"
                    >
                      <div
                        class=" css-1dimb5e-singleValue"
                      >
                        KFC
                      </div>
                      <div
                        class=" css-19bb58m"
                        data-value=""
                      >
                        <input
                          aria-activedescendant=""
                          aria-autocomplete="list"
                          aria-expanded="false"
                          aria-haspopup="true"
                          autocapitalize="none"
                          autocomplete="off"
                          autocorrect="off"
                          class=""
                          id="react-select-4-input"
                          role="combobox"
                          spellcheck="false"
                          style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                          tabindex="0"
                          type="text"
                          value=""
                        />
                      </div>
                    </div>
                    <div
                      class=" css-1wy0on6"
                    >
                      <span
                        class=" css-1u9des2-indicatorSeparator"
                      />
                      <div
                        aria-hidden="true"
                        class=" css-1xc3v61-indicatorContainer"
                      >
                        <svg
                          aria-hidden="true"
                          class="css-tj5bde-Svg"
                          focusable="false"
                          height="20"
                          viewBox="0 0 20 20"
                          width="20"
                        >
                          <path
                            d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                          />
                        </svg>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                data-testid="address-select"
                id="address"
              >
                <div
                  class=" css-b62m3t-container"
                >
                  <span
                    class="css-1f43avz-a11yText-A11yText"
                    id="react-select-5-live-region"
                  />
                  <span
                    aria-atomic="false"
                    aria-live="polite"
                    aria-relevant="additions text"
                    class="css-1f43avz-a11yText-A11yText"
                    role="log"
                  />
                  <div
                    class=" css-185hrlv-control"
                  >
                    <div
                      class=" css-hlgwow"
                    >
                      <div
                        class=" css-1jqq78o-placeholder"
                        id="react-select-5-placeholder"
                      >
                        Where should we deliver your order?
                      </div>
                      <div
                        class=" css-19bb58m"
                        data-value=""
                      >
                        <input
                          aria-activedescendant=""
                          aria-autocomplete="list"
                          aria-describedby="react-select-5-placeholder"
                          aria-expanded="false"
                          aria-haspopup="true"
                          autocapitalize="none"
                          autocomplete="off"
                          autocorrect="off"
                          class=""
                          id="react-select-5-input"
                          role="combobox"
                          spellcheck="false"
                          style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                          tabindex="0"
                          type="text"
                          value=""
                        />
                      </div>
                    </div>
                    <div
                      class=" css-1wy0on6"
                    >
                      <span
                        class=" css-1u9des2-indicatorSeparator"
                      />
                      <div
                        aria-hidden="true"
                        class=" css-1xc3v61-indicatorContainer"
                      >
                        <svg
                          aria-hidden="true"
                          class="css-tj5bde-Svg"
                          focusable="false"
                          height="20"
                          viewBox="0 0 20 20"
                          width="20"
                        >
                          <path
                            d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                          />
                        </svg>
                      </div>
                    </div>
                  </div>
                </div>
                <button
                  id="add-address"
                >
                  Add address
                </button>
              </div>
              <button
                id="complete"
              >
                Complete order
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Cart component selects restaurant to deliver from 1`] = `
<DocumentFragment>
  <div
    id="cart"
  >
    <div
      class="base-header"
    >
      <img
        alt=""
        id="load-img"
        src="test-file-stub"
      />
    </div>
    <div
      id="main"
      style=""
    >
      <div
        class="container"
        id="cart-items-list"
      >
        <div
          class="item"
        >
          <button
            class="cross"
            data-testid="cross"
          >
            <img
              alt=""
              src="test-file-stub"
            />
          </button>
          <div
            class="main"
          >
            <img
              alt=""
              src="undefined/media/img/food/SweDish.png"
            />
            <div
              class="info"
            >
              <h4>
                Swe Dish
              </h4>
              <h5>
                $19.99
              </h5>
            </div>
          </div>
          <div
            class="quant"
          >
            <button
              data-testid="minus"
            >
              <img
                alt=""
                src="test-file-stub"
              />
            </button>
            <div>
              <p>
                3
              </p>
            </div>
            <button
              data-testid="plus"
            >
              <img
                alt=""
                src="test-file-stub"
              />
            </button>
          </div>
        </div>
      </div>
      <div
        style="display: block;"
      >
        <div
          id="order"
        >
          <div
            id="costs"
          >
            <p
              class="error"
              style="display: none;"
            >
              Invalid coupon
            </p>
            <form>
              <div
                id="input"
              >
                <img
                  alt=""
                  src="test-file-stub"
                />
                <input
                  placeholder="Apply Coupon"
                  type="text"
                  value=""
                />
              </div>
              <button
                type="submit"
              >
                Apply
              </button>
            </form>
            <h6>
              Subtotal: 
              <span>
                $59.97
              </span>
            </h6>
            <h6>
              Delivery: 
              <span>
                $6.00
              </span>
            </h6>
            <h6
              style="display: none;"
            >
              Coupon:
              <span
                style="color: green;"
              >
                -$0.00
              </span>
            </h6>
            <h6
              class="total"
            >
              Total: 
              <span>
                $65.97
              </span>
            </h6>
          </div>
          <div
            id="pay"
          >
            <div
              id="userInfo"
            >
              <div
                id="address"
              >
                <div
                  id="nameDiv"
                >
                  <input
                    autocomplete="off"
                    id="name"
                    placeholder="How courier'll call you?"
                    type="text"
                    value=""
                  />
                </div>
                <div
                  id="phoneDiv"
                >
                  <input
                    autocomplete="off"
                    id="name"
                    placeholder="Phone number"
                    type="phone"
                    value="+375666999"
                  />
                </div>
              </div>
              <div
                data-testid="rest-select"
                id="restDiv"
              >
                <div
                  class=" css-b62m3t-container"
                >
                  <span
                    class="css-1f43avz-a11yText-A11yText"
                    id="react-select-16-live-region"
                  />
                  <span
                    aria-atomic="false"
                    aria-live="polite"
                    aria-relevant="additions text"
                    class="css-1f43avz-a11yText-A11yText"
                    role="log"
                  />
                  <div
                    class=" css-185hrlv-control"
                  >
                    <div
                      class=" css-hlgwow"
                    >
                      <div
                        class=" css-1dimb5e-singleValue"
                      >
                        KFC
                      </div>
                      <div
                        class=" css-19bb58m"
                        data-value=""
                      >
                        <input
                          aria-activedescendant=""
                          aria-autocomplete="list"
                          aria-expanded="false"
                          aria-haspopup="true"
                          autocapitalize="none"
                          autocomplete="off"
                          autocorrect="off"
                          class=""
                          id="react-select-16-input"
                          role="combobox"
                          spellcheck="false"
                          style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                          tabindex="0"
                          type="text"
                          value=""
                        />
                      </div>
                    </div>
                    <div
                      class=" css-1wy0on6"
                    >
                      <span
                        class=" css-1u9des2-indicatorSeparator"
                      />
                      <div
                        aria-hidden="true"
                        class=" css-1xc3v61-indicatorContainer"
                      >
                        <svg
                          aria-hidden="true"
                          class="css-tj5bde-Svg"
                          focusable="false"
                          height="20"
                          viewBox="0 0 20 20"
                          width="20"
                        >
                          <path
                            d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                          />
                        </svg>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                data-testid="address-select"
                id="address"
              >
                <div
                  class=" css-b62m3t-container"
                >
                  <span
                    class="css-1f43avz-a11yText-A11yText"
                    id="react-select-17-live-region"
                  />
                  <span
                    aria-atomic="false"
                    aria-live="polite"
                    aria-relevant="additions text"
                    class="css-1f43avz-a11yText-A11yText"
                    role="log"
                  />
                  <div
                    class=" css-185hrlv-control"
                  >
                    <div
                      class=" css-hlgwow"
                    >
                      <div
                        class=" css-1jqq78o-placeholder"
                        id="react-select-17-placeholder"
                      >
                        Where should we deliver your order?
                      </div>
                      <div
                        class=" css-19bb58m"
                        data-value=""
                      >
                        <input
                          aria-activedescendant=""
                          aria-autocomplete="list"
                          aria-describedby="react-select-17-placeholder"
                          aria-expanded="false"
                          aria-haspopup="true"
                          autocapitalize="none"
                          autocomplete="off"
                          autocorrect="off"
                          class=""
                          id="react-select-17-input"
                          role="combobox"
                          spellcheck="false"
                          style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                          tabindex="0"
                          type="text"
                          value=""
                        />
                      </div>
                    </div>
                    <div
                      class=" css-1wy0on6"
                    >
                      <span
                        class=" css-1u9des2-indicatorSeparator"
                      />
                      <div
                        aria-hidden="true"
                        class=" css-1xc3v61-indicatorContainer"
                      >
                        <svg
                          aria-hidden="true"
                          class="css-tj5bde-Svg"
                          focusable="false"
                          height="20"
                          viewBox="0 0 20 20"
                          width="20"
                        >
                          <path
                            d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                          />
                        </svg>
                      </div>
                    </div>
                  </div>
                </div>
                <button
                  id="add-address"
                >
                  Add address
                </button>
              </div>
              <button
                id="complete"
              >
                Complete order
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
